### PR TITLE
run_test.sh log function works on macOS

### PR DESCRIPTION
**Pull request by bsergean**
2 things seem to fail:

```
$ sh scripts/run_tests.sh 
-n [Mon Jun 27 10:22:24 CEST 2022] 
head: illegal byte count -- 0
```

1. The echo -n option is not supported apparently by macOS echo. Instead we can use the printf option which will not print a newline character.
2. macOS head does not like the 0 byte count for the -c option ; so we add a small test to do nothing if INDENT is still zero.

With those changes I can get run_tests.sh going.

```
[Mon Jun 27 10:26:56 CEST 2022] Starting libdeflate tests
[Mon Jun 27 10:26:56 CEST 2022] Running tests
[Mon Jun 27 10:26:56 CEST 2022]     CC=cc CFLAGS="" WRAPPER="" 
[Mon Jun 27 10:26:57 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=
[Mon Jun 27 10:27:00 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3
[Mon Jun 27 10:27:00 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3,crc32
[Mon Jun 27 10:27:01 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3,crc32,pmull
[Mon Jun 27 10:27:02 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3,crc32,pmull,neon
[Mon Jun 27 10:27:02 CEST 2022]     CC=cc CFLAGS="" WRAPPER="" FREESTANDING=1
...
```

### CODE REVIEW

File: scripts/run_tests.sh [hunk 1]
```diff
@@ -57,8 +57,8 @@ done
 INDENT=0
 
 log() {
-	echo -n "[$(date)] "
-	head -c $(( INDENT * 4 )) /dev/zero | tr '\0' ' '
+	printf "[$(date)] "
```
Comment (ebiggers) [inline scripts/run_tests.sh:60]: Shellcheck complains about thtis.  Can you use `printf "%s" "[$(date)] "` instead?

**Review by Art2539 — APPROVED** [review]

**Comment by ebiggers** [comment]
Please try it again after https://github.com/ebiggers/libdeflate/pull/212.  There were more problems that had to be fixed too.

`echo -n` works fine, though (tested on macOS 12.0.1), so I didn't change that.  Which macOS version are you using, and which version of bash?  Are you sure that `echo -n` doesn't work?  If `echo -n` does need to be replaced with `printf`, note that `gzip_tests.sh` would need to be updated too.

**Comment by bsergean** [comment]
I rebased my branch of latest master, and indeed everything works fine now, this PR wasn't relevant anymore, good that you closed it.

```
libdeflate$ bash scripts/run_tests.sh
[Sat Jul  9 23:00:15 CEST 2022] Starting libdeflate tests
[Sat Jul  9 23:00:15 CEST 2022] Running tests
[Sat Jul  9 23:00:15 CEST 2022]     CC=cc CFLAGS="" WRAPPER="" 
[Sat Jul  9 23:00:15 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=
[Sat Jul  9 23:00:18 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3
[Sat Jul  9 23:00:18 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3,crc32
[Sat Jul  9 23:00:19 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3,crc32,pmull
[Sat Jul  9 23:00:19 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3,crc32,pmull,neon
[Sat Jul  9 23:00:20 CEST 2022]     Skipping freestanding build tests due to unsupported OS
[Sat Jul  9 23:00:20 CEST 2022]     Unsupported system gzip; skipping comparison with system gzip
[Sat Jul  9 23:00:20 CEST 2022]     Running gzip program tests with CC="cc" CFLAGS=""
[Sat Jul  9 23:00:21 CEST 2022]         GZIP=/Users/benjamin.sergeant/src/foss/github/libdeflate/gzip, GUNZIP=/Users/benjamin.sergeant/src/foss/github/libdeflate/gunzip
[Sat Jul  9 23:00:28 CEST 2022] Running tests with -O3
[Sat Jul  9 23:00:28 CEST 2022]     CC=cc CFLAGS=" -O3" WRAPPER="" 
[Sat Jul  9 23:00:29 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=
[Sat Jul  9 23:00:35 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3
[Sat Jul  9 23:00:35 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3,crc32
[Sat Jul  9 23:00:36 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3,crc32,pmull
[Sat Jul  9 23:00:37 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=sha3,crc32,pmull,neon
[Sat Jul  9 23:00:37 CEST 2022]     Skipping freestanding build tests due to unsupported OS
[Sat Jul  9 23:00:37 CEST 2022]     Unsupported system gzip; skipping comparison with system gzip
[Sat Jul  9 23:00:37 CEST 2022]     Running gzip program tests with CC="cc" CFLAGS=" -O3"
[Sat Jul  9 23:00:38 CEST 2022]         GZIP=/Users/benjamin.sergeant/src/foss/github/libdeflate/gzip, GUNZIP=/Users/benjamin.sergeant/src/foss/github/libdeflate/gunzip
[Sat Jul  9 23:00:42 CEST 2022] Running tests with UBSAN
[Sat Jul  9 23:00:42 CEST 2022]     CC=cc CFLAGS=" -fsanitize=undefined -fno-sanitize-recover=undefined" WRAPPER="" 
[Sat Jul  9 23:00:44 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=
[Sat Jul  9 23:00:52 CEST 2022]     Running gzip program tests with CC="cc" CFLAGS=" -fsanitize=undefined -fno-sanitize-recover=undefined"
[Sat Jul  9 23:00:54 CEST 2022]         GZIP=/Users/benjamin.sergeant/src/foss/github/libdeflate/gzip, GUNZIP=/Users/benjamin.sergeant/src/foss/github/libdeflate/gunzip
[Sat Jul  9 23:00:58 CEST 2022] Running tests with ASAN
[Sat Jul  9 23:00:58 CEST 2022]     CC=cc CFLAGS=" -fsanitize=address -fno-sanitize-recover=address" WRAPPER="" 
[Sat Jul  9 23:01:00 CEST 2022]         Using LIBDEFLATE_DISABLE_CPU_FEATURES=
[Sat Jul  9 23:01:14 CEST 2022]     Running gzip program tests with CC="cc" CFLAGS=" -fsanitize=address -fno-sanitize-recover=address"
[Sat Jul  9 23:01:15 CEST 2022]         GZIP=/Users/benjamin.sergeant/src/foss/github/libdeflate/gzip, GUNZIP=/Users/benjamin.sergeant/src/foss/github/libdeflate/gunzip
[Sat Jul  9 23:01:28 CEST 2022] Skipping CFI tests because compiler (Apple clang version 13.1.6 (clang-1316.0.21.2.5)) doesn't support CFI
[Sat Jul  9 23:01:28 CEST 2022] Testing 'make install' and 'make uninstall'
[Sat Jul  9 23:01:28 CEST 2022]     Trying SHELL=/bin/bash
[Sat Jul  9 23:01:30 CEST 2022]     Trying SHELL=/bin/dash
[Sat Jul  9 23:01:31 CEST 2022] Checking that all global symbols are prefixed with "libdeflate_"
[Sat Jul  9 23:01:32 CEST 2022] Checking that all exported symbols are prefixed with "libdeflate"
[Sat Jul  9 23:01:32 CEST 2022] Testing USE_SHARED_LIB=1
[Sat Jul  9 23:01:40 CEST 2022] All tests passed!
```
